Controller/Comeback/Index.php

```diff
@@ -123,7 +123,7 @@ public function execute()
             ->setPath('/');
         $sectiondata = json_decode($this->cookieManager->getCookie('section_data_ids'));
```

Comment:
The object `$sectiondata` is created by the json_decode which signature is:

`json_decode(
    string $json,
    ?bool $associative = null,
    int $depth = 512,
    int $flags = 0
): [mixed](https://www.php.net/manual/en/language.types.declarations.php#language.types.declarations.mixed)`

> associative
When true, JSON objects will be returned as associative arrays; when false, JSON objects will be returned as objects. When null, JSON objects will be returned as associative arrays or objects depending on whether JSON_OBJECT_AS_ARRAY is set in the flags.

As we are not setting $associative nor $flags it will return an object not an array. Are you wure that this line 124 is not modified in your code?
As we are not passing the second parameter, `$associative`, it will be null so the $flags would be checked. AS we are not setting $flags its default value is 0 so,

Reply:
You are right, as the error message says this returns a `stdClass`, so my fix is wrong. However, the issue is still valid, since we use the latest version of the module without any modification on our side. The issue could be that the key `"cart"` is not set:

```php
$sectiondata = json_decode("{}");
if ($sectiondata) {
    $sectiondata->cart += 1000;
}
```
Output: `PHP Warning:  Undefined property: stdClass::$cart in php shell code on line 1`

Reply:
Thank you very much for you feedback!
Probably the reason for the warning is as you say that `cart` is no present in newer magento versions.
We will be releasing a new major version of the plugin in a few days. I'll close this issue and if you write to sat@sequra.com we will let you know as soon as we release the new version.